From ff0f5f8a2603b64d5d35ef348832b4a0d6b07e17 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Tue, 21 Apr 2015 16:17:36 +0800
Subject: [PATCH] MLK-10696 media: pxp-v4l2: refine the bounds sanity check
 logic

The output image area bounds check should be moved to
where the 'l', 't', 'w' and 'h' are finalized to avoid
the drect area beyond the maximum fb display area which
may cause overflow issue when calculating the PXP PS LRC
or ULC.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/media/platform/mxc/output/mxc_pxp_v4l2.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
index 780e228..9a4bd44 100644
--- a/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
+++ b/drivers/media/platform/mxc/output/mxc_pxp_v4l2.c
@@ -1034,10 +1034,6 @@ static int pxp_s_crop(struct file *file, void *fh,
 	w = max(w, PXP_MIN_PIX);
 	h = min(h, fbh);
 	h = max(h, PXP_MIN_PIX);
-	if ((l + w) > fbw)
-		l = 0;
-	if ((t + h) > fbh)
-		t = 0;
 
 	/* Round up values to PxP pixel block */
 	l = roundup(l, PXP_MIN_PIX);
@@ -1045,6 +1041,11 @@ static int pxp_s_crop(struct file *file, void *fh,
 	w = roundup(w, PXP_MIN_PIX);
 	h = roundup(h, PXP_MIN_PIX);
 
+	if ((l + w) > fbw)
+		l = 0;
+	if ((t + h) > fbh)
+		t = 0;
+
 	pxp->pxp_conf.proc_data.drect.left = l;
 	pxp->pxp_conf.proc_data.drect.top = t;
 	pxp->pxp_conf.proc_data.drect.width = w;
-- 
1.8.0

